packages/react-relay/modern/ReactRelayQueryRenderer.js

```diff
@@ -96,10 +97,26 @@ class ReactRelayQueryRenderer extends React.Component {
     };
     this._rootSubscription = null;
     this._selectionReference = null;
-    if (query) {
-      this.state = {
-        readyState: getDefaultState(),
-      };
+
+    if (operation) {
+      if (lookup && environment.check(operation.root)) {
+        // data is available in the store, render without making any requests
+        const snapshot = environment.lookup(operation.fragment);
+        this.state = {
+          readyState: {
+            error: null,
+            props: snapshot.data,
+            retry: () => {
+              this._fetch(operation, props.cacheConfig);
```

Comment:
I suppose by the time retry() is called, Flow feels that it can no longer guarantee that the value of operation didn't somehow change.